pkg/remotewrite/config.go

```diff
@@ -253,6 +284,13 @@ func GetConsolidatedConfig(jsonRawConf json.RawMessage, env map[string]string, a
 		}
 	}
 
+	envHeaders := getEnvMap(env, "K6_PROMETHEUS_HEADERS_")
+	for k, v := range envHeaders {
+		if _, ok := result.Headers[k]; !ok {
+			result.Headers[k] = v
+		}
+	}
```

Comment:
Thank you for the update! There is just one more change needed:

```suggestion
	envHeaders := getEnvMap(env, "K6_PROMETHEUS_HEADERS_")
	for k, v := range envHeaders {
		result.Headers[k] = v
	}
```
In many other cases your logic here would be just right, but this is an official k6 extension and it follows the general guideline of k6 on how config options should behave. The guideline  can be seen here:
https://k6.io/docs/using-k6/options/#using-options

> command-line flags > environment variables > exported script options > config file > defaults

So env var _should_ overwrite the values in JSON. Currently, there's only one test in `GetConsolidatedConfig` for headers and it doesn't catch this but if another test is added, e.g. with the same header name but different values in both JSON and env map, it would fail to adhere to k6 config guideline.

Reply:
After fixed, three test cases cover the priority of JSON, env and arg.